### DISCUSSION TITLE
FilterQueries typing allows using virtual properties which then cascade to joined entities if they contain a property of the same name

### DISCUSSION THREAD

**Discussion by tar672** [mikro-orm/mikro-orm#6312] [General]
### Describe the bug

if entity A has field x and entity B has an eager loaded relationship with A and a virtual field also called x then Mikro allows you to do `em.find(B, { x:"blah"})` not only will this run but the filter will be applied to A.

expected behaviours: typescript error and query returns error: no column x on table B.

### Reproduction

https://github.com/tar672/mikro-orm-filter-bug-repro

### What driver are you using?

None

### MikroORM version

6.4.2-dev.6

### Node.js version

v23.0.0

### Operating system

mac

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
This is a wontfix, you define a property on TS level, so the filters allow that. Decorator options cannot have any effect on type level, we can't just "skip non-persistent properties". Also it is valid to query by non-persistent property, mainly for internal reasons, but there are valid use cases for that too. Regarding scoping/aliasing, non-persistent properties cannot be aliased, they are not supposed to represent a real column, but an alias for something you bring in yourself (e.g. as part of a raw query).

What are you trying to do with that non-persistent property?

**Comment by tar672** [comment]
Hey @B4nan, thanks for such a quick response! decorator level typing restriction makes sense thanks for explaining.

Not trying to achieve anything just found the behaviour strange/concerning. Our example was a `status` field which is persisted on the parent entity and a virtual property on the child entities. A mistake in our code was doing
`em.find(Child, {status: blah}, {populate: ['parent']})`
which was running but was actually filtering on the Parent status property. Obviously a mistake in our code but like I said I would have expected  a `'status' does not exist on 'child_table' error`.

I wasn't aware of the use cases for querying for virtual properties though so given that I see why it isn't aliased and this is more tricky.

Great project by the way, really enjoying using it.
